### PR TITLE
SONAR-9805 allow to change the log level of a cluster

### PR DESCRIPTION
**Pull request by bartfastiel**

### CODE REVIEW

File: sonar-core/src/main/java/org/sonar/core/platform/HazelcastDistributedCallComponentContainer.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+import java.util.function.Supplier;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Helper class, that makes some components available statically, so that they can be used
+ * when a distributed call is sent to cluster members of both web and ce type.
+ */
+public class HazelcastDistributedCallComponentContainer {
```
Comment (SonarTech) [inline sonar-core/src/main/java/org/sonar/core/platform/HazelcastDistributedCallComponentContainer.java]: MAJOR Add a private constructor to hide the implicit public one. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1118)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: server/sonar-server/src/main/java/org/sonar/server/platform/monitoring/cluster/ProcessInfoProvider.java [hunk 2]
```diff
@@ -25,38 +25,37 @@
...
+  public ProcessInfoProvider(SystemInfoSection[] sections) {
     this.sections = Arrays.stream(sections)
       .filter(section -> !(section instanceof Global))
       .collect(Collectors.toList());
   }
 
   @Override
   public void start() {
-    ProcessInfoProviderRef.setProcessInfoProvider(this);
+    INSTANCE = this;
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/platform/monitoring/cluster/ProcessInfoProvider.java]: CRITICAL Make the enclosing method "static" or remove this set. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2696)
CRITICAL Synchronize this lazy initialization of 'INSTANCE' [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2444)

File: server/sonar-server/src/main/java/org/sonar/server/platform/monitoring/cluster/ProcessInfoProvider.java [hunk 3]
```diff
@@ -25,38 +25,37 @@
...
   }
 
   @Override
   public void stop() {
-    ProcessInfoProviderRef.setProcessInfoProvider(null);
+    INSTANCE = null;
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/platform/monitoring/cluster/ProcessInfoProvider.java]: CRITICAL Make the enclosing method "static" or remove this set. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2696)
CRITICAL Synchronize this lazy initialization of 'INSTANCE' [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2444)

File: server/sonar-server/src/main/java/org/sonar/server/platform/monitoring/cluster/ProcessInfoProvider.java [hunk 4]
```diff
@@ -25,38 +25,37 @@
...
 import org.sonar.process.systeminfo.SystemInfoSection;
 import org.sonar.process.systeminfo.protobuf.ProtobufSystemInfo;
 
 @ServerSide
 @ComputeEngineSide
-public class ProcessInfoProviderImpl implements ProcessInfoProvider, Startable {
+public class ProcessInfoProvider implements Startable {
 
+  /** */
+  private static ProcessInfoProvider INSTANCE;
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/platform/monitoring/cluster/ProcessInfoProvider.java]: MINOR Rename this field "INSTANCE" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3008)

File: server/sonar-server/src/main/java/org/sonar/server/platform/ServerLogging.java [hunk 5]
```diff
@@ -23,35 +23,60 @@
...
+  ServerLogging(LogbackHelper helper, Configuration config, ServerProcessLogging serverProcessLogging, Database database) {
     this.helper = helper;
     this.config = config;
+    this.serverProcessLogging = serverProcessLogging;
+    this.database = database;
+  }
+
+  @Override
+  public void start() {
+    INSTANCE = this;
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/platform/ServerLogging.java]: CRITICAL Make the enclosing method "static" or remove this set. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2696)
CRITICAL Synchronize this lazy initialization of 'INSTANCE' [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2444)

File: server/sonar-server/src/main/java/org/sonar/server/platform/ServerLogging.java [hunk 6]
```diff
@@ -23,35 +23,60 @@
...
+  }
+
+  @Override
+  public void stop() {
+    INSTANCE = null;
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/platform/ServerLogging.java]: CRITICAL Make the enclosing method "static" or remove this set. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2696)
CRITICAL Synchronize this lazy initialization of 'INSTANCE' [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2444)

File: server/sonar-server/src/main/java/org/sonar/server/platform/ServerLogging.java [hunk 7]
```diff
@@ -23,35 +23,60 @@
...
 
+import static org.sonar.api.utils.log.LoggerLevel.TRACE;
+
 @ServerSide
 @ComputeEngineSide
-public class ServerLogging {
+public class ServerLogging implements Startable {
 
+  /** Used for Hazelcast's distributed queries in cluster mode */
+  private static ServerLogging INSTANCE;
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/platform/ServerLogging.java]: MINOR Rename this field "INSTANCE" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3008)

**Comment by SonarTech** [comment]
SonarQube analysis reported 10 issues
* ![CRITICAL][CRITICAL] 8 critical
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: server/sonar-server/src/main/java/org/sonar/server/platform/ServerLogging.java (hunk 5 above)
Comment (bartfastiel) [reply server/sonar-server/src/main/java/org/sonar/server/platform/ServerLogging.java]: Won't fix.

File: server/sonar-server/src/main/java/org/sonar/server/platform/ServerLogging.java (hunk 6 above)
Comment (bartfastiel) [reply server/sonar-server/src/main/java/org/sonar/server/platform/ServerLogging.java]: Won't fix.

File: server/sonar-server/src/main/java/org/sonar/server/platform/ws/ChangeLogLevelClusterService.java [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+import org.sonar.api.utils.log.Loggers;
+import org.sonar.process.ProcessId;
+import org.sonar.process.cluster.hz.DistributedCall;
+import org.sonar.process.cluster.hz.HazelcastMember;
+import org.sonar.process.cluster.hz.HazelcastMemberSelectors;
+import org.sonar.server.platform.ServerLogging;
+
+public class ChangeLogLevelClusterService implements ChangeLogLevelService {
+
+  private static final int CLUSTER_TIMEOUT = 5000;
```
Comment (bartfastiel) [inline server/sonar-server/src/main/java/org/sonar/server/platform/ws/ChangeLogLevelClusterService.java]: Change name to include ```MS```, change type to ```long```

File: server/sonar-server/src/main/java/org/sonar/server/platform/monitoring/cluster/ProcessInfoProvider.java [hunk 9]
```diff
@@ -25,38 +25,37 @@
...
 import org.sonar.process.systeminfo.SystemInfoSection;
 import org.sonar.process.systeminfo.protobuf.ProtobufSystemInfo;
 
 @ServerSide
 @ComputeEngineSide
-public class ProcessInfoProviderImpl implements ProcessInfoProvider, Startable {
+public class ProcessInfoProvider implements Startable {
 
+  /** Used for Hazelcast's distributed queries in cluster mode */
+  private static ProcessInfoProvider INSTANCE;
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/platform/monitoring/cluster/ProcessInfoProvider.java]: MINOR Rename this field "INSTANCE" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3008)

**Pull request merged**